### ISSUE TITLE
Add Product type to SampleDataSource

### ISSUE THREAD

**Issue by LanceMcCarthy** [LanceMcCarthy/CommonHelpers#1]
There has been a feature need to also have a Product object type to better facilitate DataGrid scenarios.

This request is to add a Product object, similar to below class, to the sample data types:

```csharp
public partial class Product
{
	public int ProductId { get; set; }
	public string ProductName { get; set; }
	public int? SupplierId { get; set; }
	public int? CategoryId { get; set; }
	public string QuantityPerUnit { get; set; }
	public decimal? UnitPrice { get; set; }
	public short? UnitsInStock { get; set; }
	public short? UnitsOnOrder { get; set; }
	public short? ReorderLevel { get; set; }
	public bool Discontinued { get; set; }
}

``

**Comment by LanceMcCarthy** [comment]
Added `Product`, `Supplier` and `Category` object models, the SampleDataService supports these new models with data generation methods.

![image](https://user-images.githubusercontent.com/3520532/48663147-bab1bf00-ea59-11e8-8996-882d39abf155.png)

**Comment by LanceMcCarthy** [comment]
This is now available in v 1.0.8 and later.
